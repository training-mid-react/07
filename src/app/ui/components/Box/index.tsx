import { CELL } from "@/app/core/interfaces/game.interface";
import "./style.scss"

interface Props {
    value: CELL;
    onClick: () => void;
}

const Box = ({ value, onClick }: Props) => {
    return (
        <div data-testid="box" className="box" onClick={onClick}>
            {
                value && <div style={{backgroundColor: value.color}} className="box__chip"></div>
            }
        </div>
    );
}

export default Box;