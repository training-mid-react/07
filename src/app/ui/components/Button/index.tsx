import React from 'react';
import './style.scss';

interface ButtonProps {
  children: React.ReactNode;
  className?: string;
  onClick: () => void;
  disabled?: boolean;
}

const Button: React.FC<ButtonProps> = ({ children, onClick, disabled, className }) => {
  return (
    <button data-testid='button-component' className={className || 'custom-button'} onClick={onClick} disabled={disabled}>
      {children}
    </button>
  );
};

export default Button;
