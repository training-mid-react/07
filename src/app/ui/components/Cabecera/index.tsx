
import "./style.scss";

interface Props {
    currentPlayer: number
}

const Cabecera = ({ currentPlayer }: Props) => {
    return (
        <>
            <section className="cabecera">
                <div>
                    <h2>Jugador 1</h2>
                    <div className="ficha-cabecera ficha-cabecera--jugador1"></div>
                </div>
                <div>
                    <h2>Jugador 2</h2>
                    <div className="ficha-cabecera ficha-cabecera--jugador2"></div>
                </div>
            </section>
            <h3>Turno del jugador: {currentPlayer}</h3>
        </>

    )
}

export default Cabecera 