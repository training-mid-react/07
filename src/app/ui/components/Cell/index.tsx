import { FC } from 'react';
import { PLAYER } from '../../../../core/interfaces';
import './style.scss';

export interface CellProps {
  value: PLAYER | null;
}

export const Cell: FC<CellProps> = ({ value }) => {

  return (
    <>
    <div className={`cell cell__${value}`} />
    <div className='cell__frame'></div>
    </>
  );
};
