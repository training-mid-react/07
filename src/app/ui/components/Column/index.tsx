import './style.scss';
import { useDropCell } from '../../../core/hooks/useDropCell';

interface ColumnProps {
  columnIndex: number;
  dropPiece: (column: number) => void;
  columnData: number[];
}

function Column({ columnIndex, dropPiece, columnData }: ColumnProps) {
  const { handleColumnClick, droppingCell } = useDropCell({ columnIndex, dropPiece, columnData })
  return (
    <div className="column" onClick={handleColumnClick}>
      {columnData.map((cell, rowIndex) => {
        const isDropping = droppingCell === rowIndex;
        const cellClass = `cell ${cell === 1 ? 'cell--player1' : cell === 2 ? 'cell--player2' : ''} ${isDropping ? 'cell--dropping' : ''}`;
        return (
          <div
            key={rowIndex}
            className={cellClass}
          />
        );
      })}
    </div>
  );
};

export default Column;
