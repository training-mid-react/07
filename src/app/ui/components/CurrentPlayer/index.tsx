import './style.scss';

interface ICurrentPlayerProps {
  currentPlayer: number;
}

export const CurrentPlayer = ({ currentPlayer }: ICurrentPlayerProps) => {
  return (
    <div className="current-player">
      Es el turno del Jugador: 
      <span className={currentPlayer === 1 ? 'current-player__player1' : 'current-player__player2'}>
        {currentPlayer}
      </span>
    </div>
  );
};
