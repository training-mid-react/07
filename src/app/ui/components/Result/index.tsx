import './style.scss';

interface IResult {
    currentPlayer: number,
    winner: number | null,
    draw: boolean,
    onResetGame: () => void,
}

export const Result = ({ winner, draw, currentPlayer, onResetGame }: IResult) => {
    if (winner) {
        return (
            <div className='new-game'><button className='new-game_button' onClick={() => onResetGame()}>Jugar de nuevo</button><div className="result">¡El Ganador es el Jugador {winner}! <span className={currentPlayer === 1 ? 'result__player2' : 'result__player1'}></span> 🎉</div></div>
        )
    } if (draw && !winner) {
        return (
            <div className='new-game'><button className='new-game_button' onClick={() => onResetGame()}>Jugar de nuevo</button><div className="result">¡Es un Empate! 👌</div></div>
        )
    }
    return <div />

}