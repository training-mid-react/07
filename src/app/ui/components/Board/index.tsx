import { FC } from 'react';
import { Column } from '../Column';
import { IBoard } from '../../../../core/interfaces';
import './style.scss';

interface Props {
  board: IBoard;
  handleDrop: (colIndex: number) => void;
}

export const Board:FC<Props> = ({board, handleDrop}) => {

  return (
    <div className="board">
      {board.map((_, colIndex) => (
        <Column key={colIndex} board={board} columnIndex={colIndex} handleDrop={handleDrop} />
      ))}
      <Column key={board.length} board={board} columnIndex={board.length} handleDrop={handleDrop} />
    </div>
  );
};
