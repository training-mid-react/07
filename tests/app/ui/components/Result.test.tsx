import { render } from "@testing-library/react";
import React from "react";
import { describe, expect, test } from "vitest";
import { Result } from '../../../../src/app/ui/components/Result';

describe('Pruebas en el componente Results', () => {
  test('Match snapshot', () => {
    render(<Result currentPlayer={1} draw={false} onResetGame={() => {}} winner={1} />)
    expect(document.body).toMatchSnapshot();
  })
  test('Match snapshot', () => {
    render(<Result currentPlayer={2} draw={false} onResetGame={() => {}} winner={2} />)
    expect(document.body).toMatchSnapshot();
  })
})