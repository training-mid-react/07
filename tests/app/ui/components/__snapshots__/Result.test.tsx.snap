// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pruebas en el componente Results > Match snapshot 1`] = `
<body>
  <div>
    <div
      class="new-game"
    >
      <button
        class="new-game_button"
      >
        Jugar de nuevo
      </button>
      <div
        class="result"
      >
        Â¡El Ganador es el Jugador 
        1
        ! 
        <span
          class="result__player2"
        />
         ðŸŽ‰
      </div>
    </div>
  </div>
</body>
`;

exports[`Pruebas en el componente Results > Match snapshot 2`] = `
<body>
  <div>
    <div
      class="new-game"
    >
      <button
        class="new-game_button"
      >
        Jugar de nuevo
      </button>
      <div
        class="result"
      >
        Â¡El Ganador es el Jugador 
        2
        ! 
        <span
          class="result__player1"
        />
         ðŸŽ‰
      </div>
    </div>
  </div>
</body>
`;

exports[`Pruebas en el componente Results > Match snapshot 3`] = `
<body>
  <div>
    <div
      class="new-game"
    >
      <button
        class="new-game_button"
      >
        Jugar de nuevo
      </button>
      <div
        class="result"
      >
        Â¡El Ganador es el Jugador 
        3
        ! 
        <span
          class="result__player1"
        />
         ðŸŽ‰
      </div>
    </div>
  </div>
</body>
`;
